extra/language/c.nix

```diff
@@ -9,52 +9,67 @@ in
 with lib;
 {
   options.language.c = {
+    compiler = mkOption {
+      type = strOrPackage;
+      default = pkgs.clang;
+      defaultText = "pkgs.clang";
+      description = ''
+        Which C compiler to use.
+
+        For gcc, use pkgs.gcc-unwrapped.
```

Comment:
Having a wrapped and a unwrapped version will behave quite differently here. i.e. the wrapped version of clang will set hardening flags that the unwrapped version of gcc won't. This might be surprising for users.